modules/users/tests/server/user.server.routes.tests.js

```diff
@@ -262,6 +262,601 @@ describe('User CRUD tests', function () {
     });
   });
 
+  it('forgot password should return 400 for non-existent username', function (done) {
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: 'some_username_that_doesnt_exist'
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('No account with that username has been found');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for no username provided', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: ''
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('Username field must not be blank');
+        });
+
+        agent.post('/api/auth/forgot')
+          .expect(400)
+          .end(function (err, res) {
+            // Handle error
+            if (err) {
+              return done(err);
+            }
+
+            res.body.message.should.equal('Username field must not be blank');
+            return done();
+          });
+
+    });
+  });
+
+  it('forgot password should return 400 for non-local provider set for the user object', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: user.username
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('It seems like you signed up using your ' + user.provider + ' account');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for user password reset request', function (done) {
+    // General comment about this test-
```

Comment:
Yes. How can we sent e-mails outside? can we also receive them somehow on the test scenario to confirm it?

Also, even if we're able to set it up, it doesn't mean that on your own machine, or any other user for that matter, this kind of scenario will be supported out of the box (the email sending) and so the tests may fail there.

WDYT?